samples/gen-go/client/doer.go

```diff
@@ -91,3 +99,196 @@ func (d retryDoer) Do(c *http.Client, r *http.Request) (*http.Response, error) {
 	}
 	return resp, err
 }
+
+// circuitBreakerDoer implements the circuit breaker pattern.
+// Set debug to true to operate the circuit in a mode where it will only log
+// circuit state, and will not block any requests from going through.
+type circuitBreakerDoer struct {
+	d           doer
+	debug       bool
+	circuitName string
+}
+
+var circuitSSEOnce sync.Once
+
+// HystrixSSEEvent is emitted by hystrix-go via server-sent events. It describes
+// the state of a circuit.
+type HystrixSSEEvent struct {
+	Type                            string `json:"type"`
```

Comment:
I was worried about the case where we may want to log a different set of metrics, and how it might be useful to know what is available... For now I'll remove the ones were not using